From 9875e47079abff55f8d2c1e958e9d50de6eae7ec Mon Sep 17 00:00:00 2001
From: Peter Samuelson <peter@p12n.org>
Date: Sun, 9 Feb 2014 16:09:55 +0000
Subject: Reduce severity of "Killed by signal %d"

This produces irritating messages when using ProxyCommand or other programs
that use ssh under the covers (e.g. Subversion).  These messages are more
normally printed by the calling program, such as the shell.

According to the upstream bug, the right way to avoid this is to use the -q
option, so we may drop this patch after further investigation into whether
any software in Debian is still relying on it.

Author: Colin Watson <cjwatson@debian.org>
Bug: https://bugzilla.mindrot.org/show_bug.cgi?id=1118
Bug-Debian: http://bugs.debian.org/313371
Last-Update: 2013-09-14

Patch-Name: quieter-signals.patch
---
 clientloop.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/clientloop.c b/clientloop.c
index 73a800c..4bc5b57 100644
--- a/clientloop.c
+++ b/clientloop.c
@@ -1717,8 +1717,10 @@ client_loop(int have_pty, int escape_char_arg, int ssh2_chan_id)
 		exit_status = 0;
 	}
 
-	if (received_signal)
-		fatal("Killed by signal %d.", (int) received_signal);
+	if (received_signal) {
+		debug("Killed by signal %d.", (int) received_signal);
+		cleanup_exit((int) received_signal + 128);
+	}
 
 	/*
 	 * In interactive mode (with pseudo tty) display a message indicating
